crates/biome_js_analyze/src/lint/nursery/missing_playwright_await.rs

```diff
@@ -0,0 +1,493 @@
+
+    // Check if it's in Promise.all - if so, verify that the Promise.all call itself is properly handled
+    if let Some(promise_all_call) = find_enclosing_promise_all(call_expr) {
+        // Check if the Promise.all call is awaited or returned
+        return is_call_awaited_or_returned(&promise_all_call);
+    }
+
+    // Check if it's assigned to a variable that's later awaited
+    // This is complex and would require flow analysis, skipping for now
+
+    false
+}
+
+fn find_enclosing_promise_all(call_expr: &JsCallExpression) -> Option<JsCallExpression> {
+    let mut current = call_expr.syntax().parent();
+    
+    while let Some(node) = current {
+        // Check if we're in an array expression
+        if node.kind() == biome_js_syntax::JsSyntaxKind::JS_ARRAY_EXPRESSION {
+            // Check if the array is an argument to Promise.all
+            if let Some(parent) = node.parent() {
+                if parent.kind() == biome_js_syntax::JsSyntaxKind::JS_CALL_ARGUMENT_LIST {
+                    if let Some(call_args_parent) = parent.parent() {
+                        if call_args_parent.kind() == biome_js_syntax::JsSyntaxKind::JS_CALL_ARGUMENTS {
+                            if let Some(promise_call) = call_args_parent.parent() {
+                                if let Some(call) = JsCallExpression::cast_ref(&promise_call) {
+                                    if is_promise_all(&call) {
+                                        return Some(call);
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        
+        // Stop at function boundaries
+        if matches!(
+            node.kind(),
+            biome_js_syntax::JsSyntaxKind::JS_FUNCTION_EXPRESSION
+                | biome_js_syntax::JsSyntaxKind::JS_ARROW_FUNCTION_EXPRESSION
+        ) {
+            break;
+        }
+        
+        current = node.parent();
+    }
+    
+    None
+}
+
+fn is_promise_all(call: &JsCallExpression) -> bool {
+    if let Ok(callee) = call.callee() {
+        if let Some(member) = callee.as_js_static_member_expression() {
+            if let Ok(member_name) = member.member() {
+                if let Some(name) = member_name.as_js_name() {
+                    if let Ok(token) = name.value_token() {
+                        if token.text_trimmed() == "all" {
+                            // Check if object is Promise
+                            if let Ok(object) = member.object() {
+                                if let AnyJsExpression::JsIdentifierExpression(id) = object {
+                                    if let Ok(name) = id.name() {
+                                        if let Ok(token) = name.value_token() {
+                                            return token.text_trimmed() == "Promise";
+                                        }
+                                    }
```

Comment:
This looks an awful lot like `is_test_step_call`. It would be worth it to make a more generic function to do this check.